activerecord/test/cases/base_test.rb

```diff
@@ -1978,24 +2016,35 @@ def test_protected_environments_are_stored_as_an_array_of_string
   end
 
   test "#connected_to_many sets prevent_writes if role is reading" do
+    assert_not SecondAbstractClass.current_preventing_writes
+    assert_not ActiveRecord::Base.current_preventing_writes
+
     ActiveRecord::Base.connected_to_many([SecondAbstractClass], role: :reading) do
       assert SecondAbstractClass.current_preventing_writes
       assert_not ActiveRecord::Base.current_preventing_writes
     end
   end
 
+  test "#connected_to_many cannot be called with the reading role and prevent_writes: false" do
+    error = assert_raises ArgumentError do
+      ActiveRecord::Base.connected_to_many([SecondAbstractClass], role: :reading, prevent_writes: false) { }
+    end
+
+    assert_equal "cannot set `prevent_writes` to false when `role` is `reading`.", error.message
+  end
+
   test "#connected_to_many with a single argument for classes" do
     ActiveRecord::Base.connected_to_many(SecondAbstractClass, role: :reading) do
-      assert SecondAbstractClass.current_preventing_writes
-      assert_not ActiveRecord::Base.current_preventing_writes
+      assert SecondAbstractClass.connected_to?(role: :reading)
+      assert_not ActiveRecord::Base.connected_to?(role: :reading)
```

Comment:
There are other tests for this method that test `current_preventing_writes`, what this should probably be testing is the role connection given that is the argument provided. Same applies for the next test.